### PR TITLE
Protein Translation approaches

### PR DESCRIPTION
**Pull request by bobahop**
Approaches for Protein Translation

### CODE REVIEW

**Comment by bobahop** [comment]
@ErikSchierboom: Scala Protein Translation approaches ready for review

File: exercises/practice/protein-translation/.approaches/if-else-match-recursion/content.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  @tailrec
+  private def codonsToProteins(
+      input: String,
+      proteins: Seq[String]
+  ): Seq[String] = {
+    if (input.length < 3)
+      proteins
+    else
+      codonToProtein(input.substring(0, 3)) match {
```
Comment (ErikSchierboom) [inline exercises/practice/protein-translation/.approaches/if-else-match-recursion/content.md:20]: My Scala is quite rusty, but could you also do?

```suggestion
      codonToProtein(input.take(3)) match {
```

File: exercises/practice/protein-translation/.approaches/if-else-match-recursion/content.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        case "STOP" => proteins
+        case protein =>
+          codonsToProteins(input.substring(3), proteins :+ protein)
```
Comment (ErikSchierboom) [inline exercises/practice/protein-translation/.approaches/if-else-match-recursion/content.md:23]: Similar to the above, could this be?

```suggestion
          codonsToProteins(input.drop(3), proteins :+ protein)
```

**Comment by bobahop** [comment]
Yes, thanks, `take` and `drop` are better. Let me adjust the explanatory text...

**Review by ErikSchierboom — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
Thanks!

**Pull request merged**